exercises/chapter8/test/no-peeking/Solutions.purs

```diff
@@ -28,4 +30,34 @@ filterM f (x : xs) = do
 exceptionDivide :: Int -> Int -> Effect Int
 exceptionDivide _ 0 = throwException $ error "div zero"
 
-exceptionDivide a b = pure $ a / b
\ No newline at end of file
+exceptionDivide a b = pure $ a / b
+
+estimatePi :: Int -> Number
+estimatePi n =
+  run do
+    accRef <- new 0.0
+    for 1 (n + 1) \k ->
+      let
+        sign = if k `mod` 2 == 0 then -1.0 else 1.0
```

Comment:
Could use the `even` function https://pursuit.purescript.org/packages/purescript-integers/4.0.0/docs/Data.Int#v:even